frontend/src/views/SecretMainPage/SecretMainPage.tsx

```diff
@@ -102,6 +104,28 @@ export const SecretMainPage = () => {
     }
   });
 
+  //selectAll functionality
+  const { selectAll: selectAllSecrets } = useSelectedSecretActions();
+  const selectedSecret = useSelectedSecrets();
+
+  const areAllSecretsSelected = useMemo(() => {
```

Comment:
I don't see a reset when clicking on folder over select all

Reply:
There two areas to click on the folder row:
* anywhere except the icons: there will be an route change and on any route change the store will have an reset. 
* icons (folder icon, edit icon, delete icon): there wouldn't be any reset.

In the above comment: Do you mean that when clicking on the folder icon, there is no reset?
I think no reset on the folder icon click signifies that the folder cannot be selected on the environment page.
Reset on folder icon click might confuse the user.

Please let me know if I misunderstood your comment 🫣